### PR TITLE
DATAES-647: Use terms-query instead of multiple should-queries for In and NotIn

### PR DESCRIPTION
**Pull request by rasmusfaber**
- [X] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [X] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES-647).
- [X] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [X] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java [hunk 1]
```diff
@@ -331,6 +331,68 @@ public void shouldExecuteCustomMethodForNotIn() {
 		assertThat(page.getContent().get(0).getId()).isEqualTo(documentId2);
 	}
 
+	@Test
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java]:
```suggestion
	@Test // DATAES-647
```

File: src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java [hunk 2]
```diff
@@ -331,6 +331,68 @@ public void shouldExecuteCustomMethodForNotIn() {
...
+
+		// when
+		List<SampleEntity> list = repository.findByKeywordIn(keywords);
+
+		// then
+		assertThat(list.size()).isEqualTo(1L);
+		assertThat(list.get(0).getId()).isEqualTo(documentId1);
+	}
+
+	@Test
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java]:
```suggestion
	@Test // DATAES-647
```

File: src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java [hunk 3]
```diff
@@ -331,6 +331,68 @@ public void shouldExecuteCustomMethodForNotIn() {
 		assertThat(page.getContent().get(0).getId()).isEqualTo(documentId2);
 	}
 
+	@Test
+	public void inCanHandleManyValues() {
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java]:
```suggestion
	public void shouldHandleManyValuesQueryingIn() {
```

File: src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java [hunk 4]
```diff
@@ -331,6 +331,68 @@ public void shouldExecuteCustomMethodForNotIn() {
...
+		// when
+		List<SampleEntity> list = repository.findByKeywordIn(keywords);
+
+		// then
+		assertThat(list.size()).isEqualTo(1L);
+		assertThat(list.get(0).getId()).isEqualTo(documentId1);
+	}
+
+	@Test
+	public void notInCanHandleManyValues() {
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/repositories/custommethod/CustomMethodRepositoryBaseTests.java]:
```suggestion
	public void shouldHandleManyValuesQueryingNotIn() {
```

**Comment by rasmusfaber** [comment]
I have made the requested changes and made a new pull-request against 4.0.x: https://github.com/spring-projects/spring-data-elasticsearch/pull/315

**Comment by sothawo** [comment]
see #315

**Pull request closed without merging**